### PR TITLE
Read user should not send the password

### PR DESCRIPTION
**Pull request by xbill82**
**Warning** This PR should be merged only after the authentication plugins have been updated.

The `getCurrentUser` call in the `authController` currently sends the user's (encrypted) password through the wire. This is bad, isn't it?

### CODE REVIEW

**Comment by xbill82** [comment]
After discussion with @scottinet we decided to delegate the password deletion to the authentication plugins. To enable this, the hooks have been moved to the bottom of the read calls for users, namely

* `getCurrentUser`
* `searchUsers`
* `getUser`

**Comment by stafyniaksacha** [comment]
+1

**Comment by scottinet** [comment]
+1

**Comment by ballinette** [comment]
OK, so, before merging this, we should update https://github.com/kuzzleio/kuzzle-plugin-auth-passport-local and add a pipe on "auth:getCurrentUser" event that remove the password from the responseObject ?

**Comment by scottinet** [comment]
@ballinette > Yes. And also on `security:searchUsers` and on `security:getUser`

**Comment by ballinette** [comment]
OK, I'm on it ;)

File: lib/api/controllers/authController.js [hunk 1]
```diff
@@ -84,6 +83,9 @@ module.exports = function AuthController (kuzzle) {
         delete response._source._id;
 
         return q(new ResponseObject(requestObject, response));
+      })
+      .then(response => {
```
Comment (ballinette) [inline lib/api/controllers/authController.js]: shouldn't we better : 
* 1st call the trigger on "response"
* then return the ResponseObject with the modified response
?

I mean:
    return kuzzle.repositories.user.load(requestObject.data._id)
      .then(user => {
        var response;
(....)
        delete response._source._id;

        return  kuzzle.pluginsManager.trigger('auth:getCurrentUser', response);
      })
      .then(response => {
        return q(new ResponseObject(requestObject, response));
      });
  };

**Comment by ballinette** [comment]
@xbill82 I made the PR on plugin auth local that removes the password: https://github.com/kuzzleio/kuzzle-plugin-auth-passport-local/pull/6
... but it's necessary to fix the trigger calls according to my remark above

* then merge and the plugin PR and deploy a new plugin version
* then fix the plugin version within defaultPlugins.json

**Comment by scottinet** [comment]
Since each Kuzzle version embarks specific plugin versions, I think you can simply deploy the new plugin version, and then update the `defaultPlugins.json` on this PR.

File: lib/api/controllers/authController.js (hunk 1 above)
Comment (xbill82) [inline lib/api/controllers/authController.js]: :+1: If @scottinet is ok anyone can make the change!

**Comment by ballinette** [comment]
mmmhh changed made on triggers call...
.... but this PR will have conflicts with PR #158 from @AnthonySendra 

I suggest to cancel this one and just add a trigger on  formatUserForSerialization() function on Anthony's PR.

**Comment by xbill82** [comment]
So, should we close this PR?

**Comment by ballinette** [comment]
Yep.
The stuff it solves must be rewritten while fixing issue #159

**Pull request closed without merging**